src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -164,6 +166,27 @@ def workerCleanup(workerCleanupInfo):
         relevant information for cleaning up the worker.
         '''
         assert workerCleanupInfo.__class__.__name__ == 'WorkerCleanupInfo'
+        configWorkDir = workerCleanupInfo.workDir
+        configWorkflowID = workerCleanupInfo.workflowID
+        # If the workflow ID hasn't been set, this batch system was scheduled for shutdown before
+        # the jobstore was set up. Hence there's no need for a cleanup
+        if configWorkDir is None:
+            return
+        # This block of code will clean up after shared caching.
+        # If configWorkDir wasn't provided, then it was setup using mkdtemp in thw worker. We
+        # recreate the name using the same method.
+        if not configWorkDir:
+            configWorkDir = tempfile.mkdtemp()  # Created just to get the temp dir
+            shutil.rmtree(configWorkDir)  # Delete immediately since it's useless
+            configWorkDir = os.path.dirname(configWorkDir)
+        sharedCacheDir = os.path.join(configWorkDir, 'cache-' + configWorkflowID)
+        # The shared cache doesn't necessarily exist.
+        if os.path.exists(sharedCacheDir):
+            shutil.rmtree(sharedCacheDir)
+        # Remove temp cache dirs that didn't clean up themselves, if any
+        tempCacheDirs = [x for x in os.listdir(configWorkDir) if x.startswith('.ctmp-')]
+        for tempCacheDir in tempCacheDirs:
+            shutil.rmtree(os.path.join(configWorkDir, tempCacheDir))
```

Comment:
Currently it looks like it's specific to caching but we reasoned that it can be used for any other future cleanup operations that might be implemented. It is called upon batch system shutdown hence was put here.